Sources/SwiftCurrent_SwiftUI/Extensions/WorkflowExtensions.swift

```diff
@@ -36,14 +36,40 @@ extension Workflow where F: FlowRepresentable & View {
      - Parameter flowPersistence: a closure returning a `FlowPersistence` representing how this item in the workflow should persist.
      - Returns: a new workflow with the additional `FlowRepresentable` item.
      */
-    public func thenProceed<F: FlowRepresentable & View>(with type: F.Type,
-                                                         launchStyle: LaunchStyle = .default,
-                                                         flowPersistence: @escaping () -> FlowPersistence) -> Workflow<F> where F.WorkflowInput == Never {
-        let workflow = Workflow<F>(first)
+    public func thenProceed<FR: FlowRepresentable & View>(with type: FR.Type,
+                                                          launchStyle: LaunchStyle = .default,
+                                                          flowPersistence: @escaping (FR.WorkflowInput) -> FlowPersistence) -> Workflow<FR> where F.WorkflowOutput == FR.WorkflowInput {
+        let workflow = Workflow<FR>(first)
         workflow.append(ExtendedFlowRepresentableMetadata(flowRepresentableType: type,
-                                                          launchStyle: launchStyle) { _ in flowPersistence() })
+                                                          launchStyle: launchStyle) { data in
+            guard case.args(let extracted) = data,
+                  let cast = extracted as? FR.WorkflowInput else { return .default }
+
+            return flowPersistence(cast)
+        })
         return workflow
     }
+
+    /**
+     Adds an item to the workflow; enforces the `FlowRepresentable.WorkflowOutput` of the previous item matches the `FlowRepresentable.WorkflowInput` of this item.
+     - Parameter type: a reference to the next `FlowRepresentable`'s concrete type in the workflow.
+     - Parameter launchStyle: the `LaunchStyle` the `FlowRepresentable` should use while it's part of this workflow.
+     - Parameter flowPersistence: a closure returning a `FlowPersistence` representing how this item in the workflow should persist.
+     - Returns: a new workflow with the additional `FlowRepresentable` item.
+     */
+//    public func thenProceed<F: FlowRepresentable & View>(with type: F.Type,
+//                                                         launchStyle: LaunchStyle = .default,
+//                                                         flowPersistence: @escaping (F.WorkflowInput) -> FlowPersistence) -> Workflow<F> where F.WorkflowInput == AnyWorkflow.PassedArgs {
+//        let workflow = Workflow<F>(first)
+//        workflow.append(ExtendedFlowRepresentableMetadata(flowRepresentableType: type,
+//                                                          launchStyle: launchStyle) { data in
+//            guard case.args(let extracted) = data,
+//                  let cast = extracted as? F.WorkflowInput else { return .default }
+//
+//            return flowPersistence(cast)
+//        })
+//        return workflow
+//    }
```

Comment:
Hate to commit this much commented code, but I didn't trust myself to remember the signatures I want to drive out next over the weekend 😬.